pkg/streamingpromql/functions.go

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package streamingpromql
+
+import (
+	"fmt"
+
+	"github.com/grafana/mimir/pkg/streamingpromql/functions"
+	"github.com/grafana/mimir/pkg/streamingpromql/operators"
+	"github.com/grafana/mimir/pkg/streamingpromql/pooling"
+	"github.com/grafana/mimir/pkg/streamingpromql/types"
+)
+
+type InstantVectorFunctionOperatorFactory func(args []types.Operator, pool *pooling.LimitingPool) (types.InstantVectorOperator, error)
+
+// SingleInputVectorFunctionOperator creates an InstantVectorFunctionOperator for functions
+// that have exactly 1 argument (v instant-vector).
+//
+// Parameters:
+//   - name: The name of the function.
+//   - metadataFunc: The function for handling metadata
+//   - seriesDataFunc: The function to handle series data
+//
+// Returns:
+//
+//	An InstantVectorFunctionOperator.
```

Comment:
[nit] I'm not sure these `Returns: ...` comments are so helpful - they just tell the reader what the method signature already tells them.

If we're going to keep this, then we need to update the type name here to reflect the new naming: `InstantVectorFunctionOperatorFactory`

Reply:
(same applies below)